app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  enum role: %i[user moderator admin]
+  after_initialize :set_default_role, if: :new_record?
+
+  def set_default_role
+    role || :user
+  end
+
+  has_one_attached :avatar
+  has_many :entities
+  has_many :groups
+
+  validates :name, presence: true
+  validates :email, presence: true
```

Comment:
- [OPTIONAL] Great job adding user name validation 👏🏻 Let's now validate all user input (password, ...) to make sure that anyone with bad intentions cannot compromise their app.